proxy/envoy/config.go

```diff
@@ -184,13 +184,28 @@ func buildOutboundFilters(instances []*model.ServiceInstance, services []*model.
 	suffix := sharedInstanceHost(instances)
 	httpConfigs := make(HTTPRouteConfigs)
 
+	// get all the route rules applicable to the instances
+	rules := config.RouteRulesBySource("", instances)
```

Comment:
Where is this code expected to run? When running in a pod, this will be a single instance and not an array. This cannot run on the host as a daemon set. And even if it does, same concept applies.